stock-managment/src/main/java/org/dhis2/android/rtsm/ui/home/HomeActivity.kt

```diff
@@ -43,6 +48,7 @@ class HomeActivity : AppCompatActivity(), OnOrgUnitSelectionFinished {
     private var themeColor = R.color.colorPrimary
     private lateinit var filterManager: FilterManager
     private var orgUnitList = listOf<OrganisationUnit>()
+    private lateinit var barcodeLauncher: ActivityResultLauncher<ScanOptions>
```

Comment:
I think, that instead of a lateinit you can initialize it here:
```
private val barcodeLauncher = registerForActivityResult(
                ScanContract()
            ) { scanIntentResult ->
                if (scanIntentResult.getContents() == null) {
                    Toast.makeText(this, "Scan cancelled!", Toast.LENGTH_SHORT).show()
                } else {
                    onScanCompleted(
                        scanIntentResult
                    )
                }
            }
```
What do you think?

Reply:
Well, as the value is added on create event, I think there's no need of initializing value, because the value is assigned on create Event lifecycle. We won't set value after this event, so I think that we won't have problems of initialization. What do you think?